qiskit/quantum_info/synthesis/one_qubit_decompose.py

```diff
@@ -184,6 +187,7 @@ def basis(self, basis):
             "ZYZ": (self._params_zyz, self._circuit_zyz),
             "ZXZ": (self._params_zxz, self._circuit_zxz),
             "XYX": (self._params_xyx, self._circuit_xyx),
+            "XZX": (self._params_xzx, self._circuit_xzx),
```

Comment:
Can you add the `XZX` basis to the `ONE_QUBIT_EULER_BASIS_GATES` dictionary at the top of the file so that the `UnitarySynthesis` and `Optimize1qGatesDecomposition` passes know that it's an available option.